### PR TITLE
add mc2

### PR DESCRIPTION
**Pull request by c-cube**
To install mc2: `opam pin add mc2 https://github.com/c-cube/mc2` with a recent flambda compiler and menhir.

### CODE REVIEW

File: src/main.ml [hunk 1]
```diff
@@ -52,6 +52,25 @@ let msat =
     pre; solve;
   }
 
+(* mc² *)
+let mc2 =
+  let open Mc2_core in
+  let pre clauses =
+    let solver = Solver.create ~plugins:[Mc2_dimacs.Plugin_sat.plugin] () in
```
Comment (Gbury) [inline src/main.ml:59]: I'd prefer the solver to be create in the `solve` function, as id done for the other sat solvers used, or is it really not possible ?

**Review by Gbury — COMMENTED** [review]
Looks good, but it would be interesting to see if the solver creation is a significant cost or not, given that for other solvers, the cost of creating the solver is included in the `solve`function

File: src/main.ml (hunk 1 above)
Comment (c-cube) [reply src/main.ml:59]: This is not possible. I cannot create terms without the solver (they depend on internal plugin IDs).

**Comment by Gbury** [comment]
Looks good, ready to merge ?

**Comment by c-cube** [comment]
I think so. The small discrepancy in time required for creating the solver should be insignificant.

**Pull request merged**